jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/AsyncContextEvent.java

```diff
@@ -42,8 +42,11 @@ public AsyncContextEvent(ServletContextHandler.ServletScopedContext context, Asy
         _asyncContext = asyncContext;
         _servletContext = ServletContextHandler.getServletContext(context);
         _state = state;
-        // TODO better than this:
-        _baseURI = request == null ? null : (request instanceof HttpServletRequest hr) ? HttpURI.from(hr.getRequestURI()) : null;
+
+        if (request instanceof HttpServletRequest hr)
+            _baseURI = HttpURI.build(hr.getRequestURI()).query(hr.getQueryString());
+        else
+            _baseURI = null;
```

Comment:
absolute niggle (but do the rename at least)
```suggestion
        _baseURI =  (request instanceof HttpServletRequest hsr) ? HttpURI.build(hsr.getRequestURI()).query(hsr.getQueryString()) : null;
```